software/lib/snap.c

```diff
@@ -793,9 +794,16 @@ int snap_action_sync_execute_job(struct snap_action *action,
 	snap_action_start(action);
 	completed = snap_action_completed(action, &rc, timeout_sec);
 
+	/* Issue #360 */
+	if (rc != 0) {
+		snap_trace("%s: EIO rc=%d completed=%d\n", __func__,
+			   rc, completed);
+		rc = SNAP_EIO;
+		goto __snap_action_sync_execute_job_exit;
+	}
 	if (completed == 0) {
 		/* Not done */
-		snap_trace("%s: rc=%d completed=%d\n", __func__,
+		snap_trace("%s: ETIME rc=%d completed=%d\n", __func__,
```

Comment:
snap_trace("%s: ETIME rc=%d Not completed\n", __func__,
--- Why to print "completed" since i know its 0 !